bundles/org.eclipse.jface/src/org/eclipse/jface/viewers/AbstractTreeViewer.java

```diff
@@ -2573,24 +2689,25 @@ public int hashCode(Object element) {
 
 	/**
 	 * Sets whether the node corresponding to the given element or tree path is
-	 * expanded or collapsed.
+	 * expanded or collapsed. Will auto expand paths of single-children if setup by
+	 * {@link #setAutoExpandOnSingleChildLevel(int) }
```

Comment:
This is a breaking API change. It modifies the postcondition of a public API method, thus changes its behavior and requires a potential adaptation of all implementations (i.e., sublasses overwriting the method).
The method is, e.g., also used to perform refresh operations, which first collapse the element and then expand it again. It is unexpected for these callers (and the users of those refresh operations) that the tree items suddenty expand on refresh operations if the new auto-expand functionality is enabled.

I think we need to add a new method like `setExpandedStateWithAutoExpandOnSingleChild` providing that functionality. We then have to check the existing callers of `setExpandedState` for whether they can be should be changed to calls of the new methods. It becomes clear that adaptations of existing users may be necessary when the documentation of `setAutoExpandOnSingleChildLevel` properly states when the auto-expand mechanism will be used.